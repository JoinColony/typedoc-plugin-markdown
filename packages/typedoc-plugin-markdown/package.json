{
  "name": "typedoc-plugin-markdown",
  "version": "3.12.1",
  "description": "A plugin for TypeDoc that enables TypeScript API documentation to be generated in Markdown.",
  "main": "dist/index.js",
  "files": [
    "dist/"
  ],
  "scripts": {
    "lint": "eslint ./src --ext .ts",
    "prepublishOnly": "yarn run lint && yarn run build && && yarn run test",
    "build": "rm -rf dist && make-dir ./dist/resources && yarn run precompile && tsc",
    "test": "jest --colors --runInBand --updateSnapshot -t \"Comments:\"",
    "build-and-test": "yarn run build && yarn run test",
    "precompile": "handlebars ./src/resources/templates -c handlebars/runtime -f ./dist/resources/templates.js && handlebars ./src/resources/partials -c handlebars/runtime -p -f ./dist/resources/partials.js && node ./build/helpers.js",
    "demos": "yarn run build && yarn run demo:md && yarn run demo:html",
    "demo:md": "typedoc --plugin typedoc-plugin-markdown --options ../../test/typedoc-options.json --out ./demo/md",
    "demo:html": "typedoc --plugin none --options ../../test/typedoc-options.json --out ./demo/html"
  },
  "author": "Thomas Grey",
  "license": "MIT",
  "bugs": {
    "url": "https://github.com/tgreyuk/typedoc-plugin-markdown/issues"
  },
  "repository": {
    "type": "git",
    "url": "git+https://github.com/tgreyuk/typedoc-plugin-markdown.git",
    "directory": "packages/typedoc-plugin-markdown"
  },
  "homepage": "https://github.com/tgreyuk/typedoc-plugin-markdown/tree/master/packages/typedoc-plugin-markdown",
  "keywords": [
    "markdown",
    "typescript",
    "documentation",
    "api",
    "typedocplugin",
    "typedoc"
  ],
  "peerDependencies": {
    "typedoc": ">=0.22.0"
  },
  "dependencies": {
    "handlebars": "^4.7.7"
  }
}
