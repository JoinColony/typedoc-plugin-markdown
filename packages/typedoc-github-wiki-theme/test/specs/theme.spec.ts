import * as fs from 'fs';
import { ProjectReflection } from 'typedoc';
import { MarkdownThemeRenderContext } from 'typedoc-plugin-markdown';

describe(`Theme:`, () => {
  describe(`Exports`, () => {
    let bootstrap: {
      project: ProjectReflection;
      context: MarkdownThemeRenderContext;
      outDir: string;
    };
    test(`should write sidebar for exports'`, async () => {
      bootstrap = global.bootstrap('theme.ts', {
        options: { theme: 'github-wiki' },
        plugin: ['typedoc-plugin-markdown', './dist'],
      });
      const sidebarFile = fs.readFileSync(bootstrap.outDir + '/_Sidebar.md');
      expect(sidebarFile.toString()).toMatchSnapshot();
    });
  });

  describe(`Exports`, () => {
    let bootstrap: {
      project: ProjectReflection;
      context: MarkdownThemeRenderContext;
      outDir: string;
    };
    test(`should write sidebar for modules'`, async () => {
      bootstrap = global.bootstrap(['breadcrumbs.ts', 'theme.ts'], {
        options: { theme: 'github-wiki' },
        plugin: ['typedoc-plugin-markdown', './dist'],
      });
      const sidebarFile = fs.readFileSync(bootstrap.outDir + '/_Sidebar.md');
      expect(sidebarFile.toString()).toMatchSnapshot();
    });
  });
});
