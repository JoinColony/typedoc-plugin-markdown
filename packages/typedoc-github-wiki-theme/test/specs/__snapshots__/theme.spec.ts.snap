// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Theme: Exports should write sidebar for exports' 1`] = `
"## typedoc-plugin-markdown-test

- [Home](../wiki/Home)
- [Exports](../wiki/Exports)

### Namespaces

- [namespaceA](../wiki/namespaceA)
- [namespaceB](../wiki/namespaceB)
"
`;

exports[`Theme: Exports should write sidebar for modules' 1`] = `
"## typedoc-plugin-markdown-test

- [Home](../wiki/Home)

### Modules

- [breadcrumbs](../wiki/breadcrumbs)
- [theme](../wiki/theme)

### Namespaces

- [namespaceA](../wiki/theme.namespaceA)
- [namespaceB](../wiki/theme.namespaceB)
"
`;
